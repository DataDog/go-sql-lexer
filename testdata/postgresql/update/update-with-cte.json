{
  "input": "WITH updated AS (\n  UPDATE users SET name = 'CTE Updated' WHERE id = 6 RETURNING *\n)\nSELECT * FROM updated;",
  "outputs": [
    {
      "expected": "WITH updated AS ( UPDATE users SET name = ? WHERE id = ? RETURNING * ) SELECT * FROM updated",
      "statement_metadata": {
        "size": 17,
        "tables": [
          "users"
        ],
        "commands": [
          "UPDATE",
          "SELECT"
        ],
        "comments": [],
        "procedures": []
      }
    }
  ]
}