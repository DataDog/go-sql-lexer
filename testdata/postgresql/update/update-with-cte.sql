WITH updated AS (
  UPDATE users SET name = 'CTE Updated' WHERE id = 6 RETURNING *
)
SELECT * FROM updated;
