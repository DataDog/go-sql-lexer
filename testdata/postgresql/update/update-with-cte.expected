[
    {
        "expected": "WITH updated AS ( UPDATE users SET name = ? WHERE id = ? RETURNING * ) SELECT * FROM updated",
        "statement_metadata": {
            "size": 24,
            "tables": [
                "users",
                "updated"
            ],
            "commands": [
                "UPDATE",
                "SELECT"
            ],
            "comments": [],
            "procedures": []
        }
    }
]